reya-rust-sdk/src/http_provider.rs

```diff
@@ -242,7 +242,7 @@ impl HttpProvider {
         &self,
         signer: PrivateKeySigner,
         batch_orders: &mut Vec<data_types::BatchOrder>,
-    ) -> eyre::Result<B256> // return the transaction hash
+    ) -> eyre::Result<TransactionReceipt> // return the transaction receipt
```

Comment:
"returning the tx_receipt is fine, but in that case we don't need to do any post checking on the order batch and the logs result. This can be done by/in the calling function. Separation of concern in this case." -> yes! let's discuss